### PR TITLE
Add WGSLLanguageFeatures content

### PR DESCRIPTION
**Pull request by chrisdavidmills**
### Description

Chrome 115 adds the [`WGSLLanguageFeatures`](https://gpuweb.github.io/gpuweb/#gpuwgsllanguagefeatures) feature of WebGPU. See https://chromestatus.com/feature/5149681044160512.

This PR adds content for the following items:

* The `GPU.wgslLanguageFeatures` property
* The `WGSLLanguageFeatures` class

### CODE REVIEW

File: files/en-us/web/api/wgsllanguagefeatures/index.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+The `WGSLLanguageFeatures` object is accessed via the {{domxref("GPU.wgslLanguageFeatures")}} property.
+
+You should note that not all WGSL language extensions will be available to WebGPU in all browsers that support it. If you are hoping to take advantage of a specific extension in a WebGPU app, thorough testing is advised.
+
+{{InheritanceDiagram}}
+
+## Available features
+
+We have not listed the exact set of WGSL language extensions available to be used in WebGPU, as it will vary between implementations and physical devices, and will change over time. For a list, refer to [WGSL language extensions](https://gpuweb.github.io/gpuweb/wgsl/#language-extension) in the WGSL specification.
```
Comment (toji) [inline files/en-us/web/api/wgsllanguagefeatures/index.md:22]: This is an amusing way of saying "There are none", but I understand where it's coming from in terms of forward compatibility. :grin:

**Review by toji — COMMENTED** [review]
LGTM! It's a little unfortunate that we don't even have a speculative feature name to make better examples with, but I think this is the best that can be done with the feature considering it's lack of actual use cases at the moment.

File: files/en-us/web/api/wgsllanguagefeatures/index.md (hunk 1 above)
Comment (toji) [reply files/en-us/web/api/wgsllanguagefeatures/index.md:22]: I'm not aware of anything coming down the pipe in the really near term. Pinging @dneto0 to see if he has any estimates to give on when we might first see this used.

Comment (chrisdavidmills) [reply files/en-us/web/api/wgsllanguagefeatures/index.md:22]: Cool, thanks.

File: files/en-us/web/api/wgsllanguagefeatures/index.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+browser-compat: api.WGSLLanguageFeatures
+---
+
+{{APIRef("WebGPU API")}}{{SeeCompatTable}}
+
+The **`WGSLLanguageFeatures`** interface of the {{domxref("WebGPU API", "WebGPU API", "", "nocode")}} is a [setlike](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) object that reports the [WGSL language extensions](https://gpuweb.github.io/gpuweb/wgsl/#language-extension) supported by the WebGPU implementation.
+
+The `WGSLLanguageFeatures` object is accessed via the {{domxref("GPU.wgslLanguageFeatures")}} property.
+
+You should note that not all WGSL language extensions will be available to WebGPU in all browsers that support it. If you are hoping to take advantage of a specific extension in a WebGPU app, thorough testing is advised.
```
Comment (dipikabh) [inline files/en-us/web/api/wgsllanguagefeatures/index.md:16]: does this need to go in a `Note:` box?

```suggestion
Note that not all WGSL language extensions are available to WebGPU in all browsers that support the API. We recommend you thoroughly test the extension you choose to use in a WebGPU app.
```
Nothing's wrong with the current style, the edit suggestion is to simplify the wording and change to active voice

File: files/en-us/web/api/wgsllanguagefeatures/index.md (hunk 1 above)
Comment (dipikabh) [inline files/en-us/web/api/wgsllanguagefeatures/index.md:22]: how about the following take:

```suggestion
The available WGSL language extensions that can be used in WebGPU can vary across implementations and physical devices and may also change over time. For a complete list, refer to [WGSL language extensions](https://gpuweb.github.io/gpuweb/wgsl/#language-extension) in the WGSL specification.
```

File: files/en-us/web/api/wgsllanguagefeatures/index.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+{{InheritanceDiagram}}
+
+## Available features
+
+We have not listed the exact set of WGSL language extensions available to be used in WebGPU, as it will vary between implementations and physical devices, and will change over time. For a list, refer to [WGSL language extensions](https://gpuweb.github.io/gpuweb/wgsl/#language-extension) in the WGSL specification.
+
+## Instance properties
+
+The following properties are available to all read-only [setlike](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) objects (the below links are to the {{jsxref("Set")}} global object reference pages).
```
Comment (dipikabh) [inline files/en-us/web/api/wgsllanguagefeatures/index.md:26]:
```suggestion
The following property is available to all read-only [setlike](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) objects (the property link is to the {{jsxref("Set")}} global object reference page):
```
Can we omit this "(the below links are to the {{jsxref("Set")}} global object reference pages)"?

File: files/en-us/web/api/wgsllanguagefeatures/index.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+- {{jsxref("Set.prototype.size", "size")}} {{Experimental_Inline}}
+  - : Returns the number of values in the set.
+
+## Instance methods
+
+The following methods are available to all read-only [setlike](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) objects (the below links are to the {{jsxref("Set")}} global object reference pages).
```
Comment (dipikabh) [inline files/en-us/web/api/wgsllanguagefeatures/index.md:33]:
```suggestion
The following methods are available to all read-only [setlike](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) objects (the links below are to the {{jsxref("Set")}} global object reference pages).
```

File: files/en-us/web/api/wgsllanguagefeatures/index.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+- {{jsxref("Set.prototype.has()", "has()")}} {{Experimental_Inline}}
+  - : Returns a boolean asserting whether an element is present with the given value in the set or not.
+- {{jsxref("Set.prototype.values()", "values()")}} {{Experimental_Inline}}
+  - : Returns a new iterator object that yields the **values** for each element in the set in insertion order.
+- {{jsxref("Set.prototype.keys()", "keys()")}} {{Experimental_Inline}}
+  - : An alias for {{jsxref("Set.prototype.values()", "values()")}}.
+- {{jsxref("Set.prototype.entries()", "entries()")}} {{Experimental_Inline}}
+  - : Returns a new iterator object that contains **an array of `[value, value]`** for each element in the set, in insertion order.
+- {{jsxref("Set.prototype.forEach()", "forEach()")}} {{Experimental_Inline}}
+  - : Calls a provided callback function once for each value present in the set, in insertion order.
```
Comment (dipikabh) [inline files/en-us/web/api/wgsllanguagefeatures/index.md:44]:
```suggestion
- {{jsxref("Set.prototype.has()", "has()")}} {{Experimental_Inline}}
  - : Returns a boolean asserting whether or not an element with the given value is present in the set.
- {{jsxref("Set.prototype.values()", "values()")}} {{Experimental_Inline}}
  - : Returns a new iterator object that yields **values** for each element in the set in insertion order.
- {{jsxref("Set.prototype.keys()", "keys()")}} {{Experimental_Inline}}
  - : An alias for {{jsxref("Set.prototype.values()", "values()")}}.
- {{jsxref("Set.prototype.entries()", "entries()")}} {{Experimental_Inline}}
  - : Returns a new iterator object that contains **an array of `[value, value]`** for each element in the set in insertion order.
- {{jsxref("Set.prototype.forEach()", "forEach()")}} {{Experimental_Inline}}
  - : Calls the provided callback function once for each value present in the set in insertion order.
```

File: files/en-us/web/api/gpu/wgsllanguagefeatures/index.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  - experimental
+browser-compat: api.GPU.wgslLanguageFeatures
+---
+
+{{APIRef("WebGPU API")}}{{SeeCompatTable}}
+
+The **`wgslLanguageFeatures`** read-only property of the
+{{domxref("GPU")}} interface returns a {{domxref("WGSLLanguageFeatures")}} object that reports the [WGSL language extensions](https://gpuweb.github.io/gpuweb/wgsl/#language-extension) supported by the WebGPU implementation.
+
+You should note that not all WGSL language extensions will be available to WebGPU in all browsers that support it. If you are hoping to take advantage of a specific extension in a WebGPU app, thorough testing is advised.
```
Comment (dipikabh) [inline files/en-us/web/api/gpu/wgsllanguagefeatures/index.md:16]: same comment as on the other page: Web/API/WGSLLanguageFeatures

File: files/en-us/web/api/wgsllanguagefeatures/index.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+{{Specifications}}
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- The [WebGPU API](/en-US/docs/Web/API/WebGPU_API)
```
Comment (dipikabh) [inline files/en-us/web/api/wgsllanguagefeatures/index.md:79]:
```suggestion
- [WebGPU API](/en-US/docs/Web/API/WebGPU_API)
```

File: files/en-us/web/api/gpu/wgsllanguagefeatures/index.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+{{Specifications}}
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- The [WebGPU API](/en-US/docs/Web/API/WebGPU_API)
```
Comment (dipikabh) [inline files/en-us/web/api/gpu/wgsllanguagefeatures/index.md:55]:
```suggestion
- [WebGPU API](/en-US/docs/Web/API/WebGPU_API)
```

**Review by dipikabh — CHANGES_REQUESTED** [review]
I've added some edit suggestions

File: files/en-us/web/api/wgsllanguagefeatures/index.md (hunk 2 above)
Comment (chrisdavidmills) [reply files/en-us/web/api/wgsllanguagefeatures/index.md]: OK, I accepted your change and did a little bit more tweaking in my next commit, plus I've put it in a note box and make it consistent across both docs.

File: files/en-us/web/api/wgsllanguagefeatures/index.md (hunk 1 above)
Comment (chrisdavidmills) [reply files/en-us/web/api/wgsllanguagefeatures/index.md]: Again, committed but then tweaked. I thought it was still important to still say that we made a conscious decision to not list them here.

File: files/en-us/web/api/wgsllanguagefeatures/index.md (hunk 3 above)
Comment (chrisdavidmills) [reply files/en-us/web/api/wgsllanguagefeatures/index.md:26]: Yeah, I ended up using pretty much your wording, and getting rid of the parens. they were just awkward, and didn't really add much.

File: files/en-us/web/api/gpu/wgsllanguagefeatures/index.md (hunk 6 above)
Comment (chrisdavidmills) [reply files/en-us/web/api/gpu/wgsllanguagefeatures/index.md:16]: Updated, as per above comment

**Comment by chrisdavidmills** [comment]
@dipikabh comments worked through, and changes made. Thanks for the review!

File: files/en-us/web/api/wgsllanguagefeatures/index.md (hunk 2 above)
Comment (dipikabh) [reply files/en-us/web/api/wgsllanguagefeatures/index.md]: Looks good, thanks

File: files/en-us/web/api/wgsllanguagefeatures/index.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+{{InheritanceDiagram}}
+
+## Available features
+
+The available WGSL language extensions can vary across implementations and physical devices and may also change over time; we have therefore not listed them here. For a complete list, refer to [WGSL language extensions](https://gpuweb.github.io/gpuweb/wgsl/#language-extension) in the WGSL specification.
+
+## Instance properties
+
+The following property is available to all read-only [setlike](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) objects.
```
Comment (dipikabh) [inline files/en-us/web/api/wgsllanguagefeatures/index.md:26]:
```suggestion
The following property is available to all read-only [setlike](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) objects:
```

File: files/en-us/web/api/wgsllanguagefeatures/index.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+- {{jsxref("Set.prototype.size", "size")}} {{Experimental_Inline}}
+  - : Returns the number of values in the set.
+
+## Instance methods
+
+The following methods are available to all read-only [setlike](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) objects.
```
Comment (dipikabh) [inline files/en-us/web/api/wgsllanguagefeatures/index.md:33]:
```suggestion
The following methods are available to all read-only [setlike](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) objects:
```

**Review by dipikabh — APPROVED** [review]
Thanks for the updates! One small nit about adding a colon in two places, otherwise good to go 👍
